### PR TITLE
AB#68 Add endpoint for searching tags

### PR DESCRIPTION
**Pull request by kikz0rsk**

### CODE REVIEW

File: Services/TagService.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+        public TagService(IGraphClient graphClient)
+        {
+            _graphClient = graphClient;
+        }
+
+        public async Task<TagSearchResponseDTO> SearchTagsAsync(TagSearchRequestDTO request)
+        {
+            var query = _graphClient.Cypher
+                .Match("(tag:Tag)")
```
Comment (Kesuera) [inline Services/TagService.cs:22]: Please add Tag to Constants/NodeNames and use it when creating a query

File: Services/TagService.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+                .WhereIf<TagNode>(request.Name != null && request.Name != "", (tag) => tag.Type.Contains(request.Name));
```
Comment (Kesuera) [inline Services/TagService.cs:23]: You can simplify this check using string.IsNullOrEmpty(request.Name) or even string.IsNullOrWhiteSpace(request.Name)

File: Services/TagService.cs (hunk 1 above)
Comment (kikz0rsk) [reply Services/TagService.cs]: fixed

File: Services/TagService.cs (hunk 2 above)
Comment (kikz0rsk) [reply Services/TagService.cs]: fixed

**Pull request merged**